diff --git a/pyscf/lib/ao2mo/test/test_ao2mo.py b/pyscf/lib/ao2mo/test/test_ao2mo.py
index 4bfc6b2a1..071d4aeea 100644
--- a/pyscf/lib/ao2mo/test/test_ao2mo.py
+++ b/pyscf/lib/ao2mo/test/test_ao2mo.py
@@ -26,7 +26,7 @@ from pyscf import ao2mo
 from pyscf.ao2mo import _ao2mo
 from pyscf.scf import _vhf
 
-libao2mo1 = lib.load_library('libao2mo')
+from pyscf.ao2mo._ao2mo import libao2mo as libao2mo1
 
 mol = gto.Mole()
 mol.verbose = 0
diff --git a/pyscf/lib/gto/test/test_gridao.py b/pyscf/lib/gto/test/test_gridao.py
index 5e6e63fbb..6a49862d0 100644
--- a/pyscf/lib/gto/test/test_gridao.py
+++ b/pyscf/lib/gto/test/test_gridao.py
@@ -19,7 +19,7 @@ import numpy
 from pyscf import lib, gto, df
 from pyscf.gto.eval_gto import BLKSIZE
 
-libcgto = lib.load_library('libdft')
+from pyscf.dft.numint import libdft as libcgto
 
 mol = gto.M(atom='''
             O 0.5 0.5 0.
diff --git a/pyscf/lib/vhf/test/test_nr_direct.py b/pyscf/lib/vhf/test/test_nr_direct.py
index 0e5894a25..e9f714d28 100644
--- a/pyscf/lib/vhf/test/test_nr_direct.py
+++ b/pyscf/lib/vhf/test/test_nr_direct.py
@@ -23,7 +23,7 @@ from pyscf import scf
 from pyscf import gto
 from pyscf import ao2mo
 
-libcvhf2 = lib.load_library('libcvhf')
+from pyscf.scf._vhf import libcvhf as libcvhf2
 
 mol = gto.Mole()
 mol.verbose = 0
@@ -68,7 +68,7 @@ def runjk(dm1, ncomp, intorname, filldot, *namejk):
     dmsptr = (ctypes.c_void_p*(njk*n_dm))()
     vjkptr = (ctypes.c_void_p*(njk*n_dm))()
     for i, symb in enumerate(namejk):
-        f1 = ctypes.c_void_p(_ctypes.dlsym(libcvhf2._handle, symb))
+        f1 = ctypes.cast(getattr(libcvhf2, symb), ctypes.c_void_p)
         for j in range(n_dm):
             dmsptr[i*n_dm+j] = dm1[j].ctypes.data_as(ctypes.c_void_p)
             vjkptr[i*n_dm+j] = vjk[i,j*ncomp].ctypes.data_as(ctypes.c_void_p)
